cli/azd/cmd/restore.go

```diff
@@ -118,7 +121,12 @@ func (r *restoreAction) Run(ctx context.Context) (*actions.ActionResult, error)
 		installMsg := fmt.Sprintf("Installing dependencies for %s service...", svc.Name)
 		spinner := spin.NewSpinner(r.console.Handles().Stdout, installMsg)
 		if err := spinner.Run(func() error {
-			return svc.Restore(ctx, r.env, r.commandRunner)
+			restoreTask := r.serviceManager.Restore(ctx, svc)
+			_, err := restoreTask.Await()
```

Comment:
there's no `go routine` here to get and digest the messages from the restoreTask.  I assume there are no messages sent through the channel.

Reply:
> Thanks for catching.

But there's no way to enforce that. That's the problem :(

Reply:
Again, this is the side effects of using go channels.  Go will automatically block until the channel values are consumed.  We could change this to be of an event subscription but shouldn't be required since this is an internal operation.